pyCGM_Single/Pipelines.py

```diff
@@ -182,6 +356,49 @@ def segmentFinder(key,data,targetDict,segmentDict,j,missings):
                         
 
 def rigid_fill(Data,static):
+    """Fills gaps in motion capture data.
+
+    Estimates marker positions from previous marker positions
+    or static data to fill in gaps in `Data`.
+
+    Parameters
+    ----------
+    Data : array
+        Array of dictionaries of marker data.
+    static : dict
+        Dictionary of marker data corresponding to a static trial.
+
+    Returns
+    -------
+    data : array
+        Array of dictionaries of marker data after gap filling is done.
+
+    Examples
+    --------
+    >>> from .pyCGM_Helpers import getfilenames
+    >>> from .pyCGM import pelvisJointCenter
+    >>> from numpy import array, nan, around
+    >>> from .pycgmIO import loadData, dataAsDict
+    >>> dynamic_trial,static_trial,_,_,_ = getfilenames(x=3)
+    >>> motionData = loadData(dynamic_trial)
+    SampleData/Sample_2/RoboWalk.c3d
+    >>> staticData = loadData(static_trial)
+    SampleData/Sample_2/RoboStatic.c3d
+
+    Sacrum must be calculated for this file using ``pyCGM.pelvisJointCenter``.
+
+    >>> for frame in motionData:
+    ...     frame['SACR'] = pelvisJointCenter(frame)[2]
+
+    Testing gap filling.
+
+    >>> Data = dataAsDict(motionData,npArray=True)
+    >>> Data['LFHD'][2] = array([nan, nan, nan]) #clear one frame to test gap filling
+    >>> static = dataAsDict(staticData,npArray=True)
+    >>> data = rigid_fill(Data, static)
```

Comment:
It would be helpful to see what the data was at the missing frame before and after the fill.

Reply:
I changed the doctests in this function and in several other functions in Pipelines to show data before and after gap filling.